reducers/views/channel_sidebar.ts

```diff
@@ -75,8 +75,62 @@ export function newCategoryIds(state: string[] = [], action: GenericAction): str
     }
 }
 
+export function selectedChannelIds(state: string[] = [], action: GenericAction): string[] {
+    switch (action.type) {
+    case ActionTypes.MULTISELECT_CHANNEL:
+        // Task was not previously selected
+        // now will be the only selected item
+        if (!state.includes(action.data)) {
+            return [action.data];
+        }
+
+        // Task was part of a selected group
+        // will now become the only selected item
+        if (state.length > 1) {
+            return [action.data];
+        }
+
+        // task was previously selected but not in a group
+        // we will now clear the selection
+        return [];
+    case ActionTypes.MULTISELECT_CHANNEL_ADD:
+        // if not selected - add it to the selected items
+        if (state.indexOf(action.data) === -1) {
```

Comment:
The state of this could be a javascript Set so then we don't have to do the checks

Reply:
This feels like something we could actually change across the board if it makes sense. Want to talk about it in hangouts or something?

Reply:
I'm trying to remember if we use Set or Map anywhere else in the redux code. I think we might have tried in redux or the mobile app, but we ran into issues serializing the redux state for storage on the mobile app.

Another thing to consider is that I think it's easier to treat literal arrays/objects as "immutable" than it is with Set since you can just do things like `[...arr, 123]` to add an item to an array, but it's a few more steps to do "immutably" with a Set.